File: Step-01-Computer-Science-Fundamentals/Step-02-Programming-Languages/word_count.py
# Step 02: Polyglot Architecture
# Interpreted Language (Python) Implementation

def count_words():
    try:
        # Python abstracts away the manual stream handling
        with open("sample.txt", "r") as file:
            words = file.read().split()
            print(f"[Python Interpreted Execution] Total words counted: {len(words)}")
    except FileNotFoundError:
        print("[ERROR] Unable to open the file.")

if __name__ == "__main__":
    count_words()

File: Step-02-Programming-Languages/word_count.cpp
#include <iostream>
#include <fstream>
#include <string>

// Step 02: Polyglot Architecture
// Compiled Language (C++) Implementation

int main() {
    std::ifstream file("sample.txt");
    std::string word;
    int count = 0;

    if (file.is_open()) {
        // C++ requires explicit stream handling
        while (file >> word) {
            count++;
        }
        file.close();
        std::cout << "[C++ Compiled Execution] Total words counted: " << count << std::endl;
    } else {
        std::cout << "[ERROR] Unable to open the file." << std::endl;
    }

    return 0;
}
